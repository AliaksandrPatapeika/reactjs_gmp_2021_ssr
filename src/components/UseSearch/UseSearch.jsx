import './UseSearch.less';

import React from 'react';
import {useSelector} from 'react-redux';

const UseSearch = () => {
  const activeModalWindow = useSelector((state) => state.movie.activeModalWindow);

  return !activeModalWindow && (
  <div className="useSearchContainer">
    <h1 className="title">Use Search to find movies</h1>
  </div>
  );
};

export default UseSearch;
