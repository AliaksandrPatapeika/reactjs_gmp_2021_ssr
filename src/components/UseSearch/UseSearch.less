@import "../../less/variables";

.useSearchContainer {
	display: flex;
	justify-content: center;
	background-color: @black;
	align-items: center;
}
