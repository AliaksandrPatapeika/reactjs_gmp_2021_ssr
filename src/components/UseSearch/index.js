import UseSearch from './UseSearch';

export default UseSearch;
