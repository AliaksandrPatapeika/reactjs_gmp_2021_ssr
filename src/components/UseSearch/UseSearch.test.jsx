import React from 'react';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';

import UseSearch from './UseSearch';

describe('UseSearch', () => {
  test('renders correctly', () => {
    const mockStore = configureMockStore();
    const initialState = {
      movie: {
        activeModalWindow: false
      }
    };
    const store = mockStore(initialState);
    const useSearch = renderer
      .create(
        <Provider store={store}>
          <UseSearch />
        </Provider>
      )
      .toJSON();

    expect(useSearch).toMatchSnapshot();
  });
});
