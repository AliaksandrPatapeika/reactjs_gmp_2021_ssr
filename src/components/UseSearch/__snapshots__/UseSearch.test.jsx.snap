// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UseSearch renders correctly 1`] = `
<div
  className="useSearchContainer"
>
  <h1
    className="title"
  >
    Use Search to find movies
  </h1>
</div>
`;
